BedrockBlockingCommandQueue.cpp

```diff
@@ -0,0 +1,13 @@
+#include <BedrockBlockingCommandQueue.h>
+
+void BedrockBlockingCommandQueue::startTiming(unique_ptr<BedrockCommand>& command) {
+    command->startTiming(BedrockCommand::QUEUE_BLOCKING);
+}
+
+void BedrockBlockingCommandQueue::stopTiming(unique_ptr<BedrockCommand>& command) {
+    command->stopTiming(BedrockCommand::QUEUE_BLOCKING);
+}
+
+BedrockBlockingCommandQueue::BedrockBlockingCommandQueue() :
+  BedrockCommandQueue(function<void(unique_ptr<BedrockCommand>&)>(startTiming), function<void(unique_ptr<BedrockCommand>&)>(stopTiming))
```

Comment:
Man this is too much C++ magic for me... how does this work?

Reply:
My understanding is:

`function<A(B...)>` is a template class that allow us to handle functions as object. The `A` is the return type of the function and `B` are the parameters.

In this case, for `stopTiming` and `stopTiming`, `A` is `void` and `B` is `unique_ptr<BedrockCommand>&`

Then, calling `function<A<B>(startTiming)` is how you get a reference of your `startTiming` function and then we can pass it as parameter to other functions (or constructor here) like if it was any object.

Reply:
I am going to pretend I understood that 😂 :ohnothing: